packages/wonder-blocks-button/components/button-core.js

```diff
@@ -104,48 +86,29 @@ export default class ButtonCore extends React.Component<Props> {
                         style={sharedStyles.icon}
                     />
                 )}
-                {children}
+                <View style={sharedStyles.parentWidth}>{children}</View>
```

Comment:
This would translate to: `button > div > children`, right?

If that's the case, I'd suggest to remove `View` here and try to use a different alternative.

Reply:
It would unfortunately, but how can I apply this style without View? The button can't be directly styled anymore because the StyledButton is being generated by Clickable and not ButtonCore anymore

Reply:
You could create a `StyledSpan` with `addStyle`.

Reply:
I’ve changed both Views to Spans in the latest commit

Reply:
Thanks for the change, but now that I see the working example it looks that you are adding an extra element here. What's the reason for adding `parentWidth` here?

Reply:
Without the parentWidth here the content can’t be centered 

![image](https://user-images.githubusercontent.com/22527952/60912452-34d9c280-a23a-11e9-8f15-12ae7c65c4bf.png)

Reply:
Ohh, so now we are using 3 spans inside the `<button>`.
https://github.com/Khan/wonder-blocks/pull/445/files#diff-c74745c6ff6054372236d55d0a92bf2bR97

I'm starting to feel this refactor needs to take a different approach:

Before this refactor: `button > span > label`

After this refactor: `button > span > span > span > label`

Reply:
Yeah that’s correct, I agree this isn’t ideal and I’m open to any other ideas, but this is the only way I was able to get the styling consistent with the original

Reply:
Got it, thanks for sharing! let me think about any possible alternatives and will comment back here.

Reply:
We may decide that it doesn't actually make sense to use `Clickable` internally.  I thought it would be nice too dedupe things, but if it has adverse side-effects then it may not be worth it.

Reply:
Yes, it's great to abstract that logic away from Button, but I think that for this particular component, this is causing (for now) adding some non-semantic HTML.

Reply:
To give some more context to this, this is what the W3 validator throws:
```
Error: Element div not allowed as child of element button in this context.
...
Content model for element button:
Phrasing content, but there must be no interactive content descendant.
```

This means that only [phrasing content](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Content_categories#Phrasing_content) is allowed inside buttons. Hope this helps :)